skip shared files when adding recovery key

butonic opened the pull request:
Skip shared files when adding recovery key.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/encryption?pullRequest=22) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/encryption?pullRequest=22) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/encryption?pullRequest=22) it.</sub>

PVince81 commented on lib/Recovery.php:
```diff
@@ -221,6 +221,11 @@ private function addRecoveryKeys($path) {
 		foreach ($dirContent as $item) {
 			$filePath = $item->getPath();
+			// only add key for files owned by the user
+			if($item->getStorage()->instanceOfStorage('OC\Files\Storage\Shared')) {
```
`ISharedStorage` to also exclude incoming federated shares.

See and also please review: https://github.com/owncloud/core/pull/29724

tomneedham replied:
And we can use ::class`

butonic replied:
hedcoding because we dont want inter app dependencies. might give problems during class loading in case files_sharing is disabled

PVince81 commented:
@sharidas please take over

butonic committed:
Skip shared files when adding recovery keys

This change set skips federated and normal file
share when recovery key is enabled

Signed-off-by: Sujith H <sharidasan@owncloud.com>

sharidas committed:
Unit test for shared storage when recovery is enabled

Add unit test for shared storage when recovery
is enabled

Signed-off-by: Sujith H <sharidasan@owncloud.com>

sharidas commented:
I have updated the PR by adding a unit test. I have verified this PR with normal share (i.e, sharing a file/folder within the same oC instance) and federated share. The verification was done to see the code flow. And in both cases when the recovery is enabled for user, it was hitting code https://github.com/owncloud/encryption/pull/22/files#diff-af3b60507c206ac40d8ec514b7cb4f63R344.

Ran the unit test locally and it did not caused any issue.

PVince81 reviewed (approved):
:+1:

phil-davis commented:
Backport to core `stable10` is PR https://github.com/owncloud/core/pull/34506
